In [114]:
# importing the required library 

import numpy as np
import pandas as pd


In [115]:
# loading the dataset to pandas
data=pd.read_csv('Netflix_stock_data.csv')
data

,Date,Close,High,Low,Open,Volume
0,2002-05-23,0.119643,0.124286,0.114571,0.115643,1047900000
1,2002-05-24,0.121000,0.122500,0.119714,0.121429,111048000
2,2002-05-28,0.115714,0.123214,0.115714,0.121357,66094000
3,2002-05-29,0.110357,0.116429,0.108571,0.116429,67578000
4,2002-05-30,0.107143,0.110786,0.107143,0.110786,101542000
...,...,...,...,...,...,...
5954,2026-01-22,83.540001,85.099998,82.980003,85.019997,69113300
5955,2026-01-23,86.120003,86.300003,83.279999,83.430000,64520400
5956,2026-01-26,85.699997,86.940002,85.349998,86.919998,40941900
5957,2026-01-27,85.580002,85.599998,83.879997,85.279999,37790800


🔵 Section 1: Data Understanding (Questions 1–8)

Q1. What is the shape of the dataset — how many rows and columns does it have?



In [116]:
data.shape

(5959, 6)

the data have 5959 rows and 6 columns

Q2. What are the column names and what does each column represent? (Date, Open, High, Low, Close, Volume)

In [117]:
data.columns

Index(['Date', 'Close', 'High', 'Low', 'Open', 'Volume'], dtype='object')

above columns of the dataset represent the following things:

Date= date on which the price details where recorded

open= opening price of the stock

close= closing price of the stock 

high= highest price of the stock 

low = lowest price of the stock

volumn= no of stock traded in a day



Q3. What are the data types of each column — are they correct? (Hint: check if Date is string or datetime)

In [118]:
data.dtypes

Date       object
Close     float64
High      float64
Low       float64
Open      float64
Volume      int64
dtype: object

No they are not correct. Date column have object data type(i.e. string). we can not sort or arrange the string. so it is not correct

chage the string data type of the date column into dataTime?

In [119]:
data['Date']=pd.to_datetime(data['Date'])
data.dtypes

Date      datetime64[ns]
Close            float64
High             float64
Low              float64
Open             float64
Volume             int64
dtype: object

Q4. What is the date range of this dataset — from what year to what year does the data span?


In [120]:
min_date=data.min()
print(min_date)


Date      2002-05-23 00:00:00
Close                0.037286
High                 0.041071
Low                  0.034643
Open                 0.037786
Volume                2856000
dtype: object


In [121]:
print(data['Date'])

0      2002-05-23
1      2002-05-24
2      2002-05-28
3      2002-05-29
4      2002-05-30
          ...    
5954   2026-01-22
5955   2026-01-23
5956   2026-01-26
5957   2026-01-27
5958   2026-01-28
Name: Date, Length: 5959, dtype: datetime64[ns]


In [122]:
start_date=data['Date'].min()
end_date=data['Date'].max()
print(f"starting Date = {start_date} and end Date = {end_date}")

starting Date = 2002-05-23 00:00:00 and end Date = 2026-01-28 00:00:00


Q5. How many years of Netflix stock data does this dataset cover?


In [123]:
start_year=data['Date'].dt.year.min()
end_year=data['Date'].dt.year.max()
no_of_years=end_year-start_year
print(no_of_years)

24


this dataset have 24 years of Netflix stock data


Q6. Are there any missing values in any column? If yes, how many?


In [124]:
data.isnull().sum()

Date      0
Close     0
High      0
Low       0
Open      0
Volume    0
dtype: int64

No, there are no missing values in any columns.

Q7. Are there any duplicate rows or duplicate dates in the dataset?


In [125]:
print(data.duplicated().sum())

0


So there are 0 duplicate rows in the dataset

Q8. What are the minimum, maximum, mean, and median values of the Close price column?


In [126]:
data['Close'].describe()

count    5959.000000
mean       19.793527
std        27.840294
min         0.037286
25%         0.440571
50%         5.074286
75%        33.754499
max       133.912994
Name: Close, dtype: float64

minimum Close price = 0.037286

maximum close price = 133.912994

mean or average close price = 19.793527

median close price = 5.074286

🟡 Section 2: Data Manipulation (Questions 9–17)

Q9. Convert the Date column from string type to datetime format using pd.to_datetime() — then set it as the index.

In [129]:
data['Date']=pd.to_datetime(data['Date']) # change the Date column into dateTime datatype
print(data['Date'].dtype)
print()
data.set_index('Date', inplace=True)
data.head()

datetime64[ns]



,Close,High,Low,Open,Volume
Date,,,,,
2002-05-23,0.119643,0.124286,0.114571,0.115643,1047900000
2002-05-24,0.121000,0.122500,0.119714,0.121429,111048000
2002-05-28,0.115714,0.123214,0.115714,0.121357,66094000
2002-05-29,0.110357,0.116429,0.108571,0.116429,67578000
2002-05-30,0.107143,0.110786,0.107143,0.110786,101542000


Q10. Extract new columns from Date — create separate columns for Year, Month, Day, and Day of Week.


In [138]:
data['Year']=data.index.year
data['Month']=data.index.month
data['Day']=data.index.day
data['Day of Week']= data.index.day_name()
data.head()

,Close,High,Low,Open,Volume,Year,Month,Day,Day of Week
Date,,,,,,,,,
2002-05-23,0.119643,0.124286,0.114571,0.115643,1047900000,2002,5,23,Thursday
2002-05-24,0.121000,0.122500,0.119714,0.121429,111048000,2002,5,24,Friday
2002-05-28,0.115714,0.123214,0.115714,0.121357,66094000,2002,5,28,Tuesday
2002-05-29,0.110357,0.116429,0.108571,0.116429,67578000,2002,5,29,Wednesday
2002-05-30,0.107143,0.110786,0.107143,0.110786,101542000,2002,5,30,Thursday
